File: Leetcode/338/338.cpp
class Solution {
public:
    // http://graphics.stanford.edu/~seander/bithacks.html#CountBitsSet64
    int f(int v) {
        int c;
        c =  ((v & 0xfff) * 0x1001001001001ULL & 0x84210842108421ULL) % 0x1f;
        c += (((v & 0xfff000) >> 12) * 0x1001001001001ULL & 0x84210842108421ULL) % 
             0x1f;
        c += ((v >> 24) * 0x1001001001001ULL & 0x84210842108421ULL) % 0x1f;
        return c;
    }
    vector<int> countBits(int num) {
        vector<int> result(num + 1);
        for (int i = 0; i <= num; ++i) {
            result.at(i) = f(i);
        }
        return result;
    }
};

File: Leetcode/338/338.go
func countBits(n int) []int {
	// dp(0) = 0
	// dp(1) = 1
	// dp(n) = dp(n&(n-1)) + 1
	dp := make([]int, n+1)
	dp[0] = 0
	if n >= 1 {
		dp[1] = 1
		for i := 2; i <= n; i++ {
			dp[i] = dp[i&(i-1)] + 1
		}
	}
	return dp
}